fuzzers/iso8601_fuzzer.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "util.h"
+#include "iso8601.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  // Ensure we have enough data.
+  if (size < 10) {
+    return 0;
+  }
+  char *ns = malloc(size+1);
+  memcpy(ns, data, size);
+  ns[size] = '\0';
+
+  crm_time_parse_period(ns);
+  pcmk__time_hr_new(ns);
```

Comment:
Generally these would not take the same argument -- the first parses ISO 8601 period specifications (and returns a `crm_time_period_t *` that should be freed with `crm_time_free_period()`), and the second parses ISO 8601 date/time specifications (and returns a `pcmk__time_hr_t *` that should be freed with `pcmk__time_hr_free()`).

Since you're not getting the return values, this doesn't really do anything, but of course that's fine to just exercise the code.